apps/extension/src/background/keyring/crypto.ts

```diff
@@ -0,0 +1,65 @@
+import { AES, Argon2, Argon2Params, ByteSize, Rng, Salt } from "@anoma/crypto";
+import { Bip44Path, KdfType, KeyStore, KeyStoreType } from "./types";
+
+type CryptoArgs = {
+  alias?: string;
+  address: string;
+  path: Bip44Path;
+  id: string;
+  password: string;
+  text: string;
+  type: KeyStoreType;
+};
+
+export class Crypto {
+  public encrypt(args: CryptoArgs): KeyStore {
+    const { address, alias, path, id, password, text, type } = args;
+    const salt = Salt.generate().as_string();
+    const argon2 = new Argon2(password, salt);
+    const { params, key } = argon2.to_serialized();
+
+    const iv = Rng.generate_bytes(ByteSize.N12);
+    const aes = new AES(key, iv);
+    const encrypted = aes.encrypt(text);
+    aes.free();
+
+    return {
+      alias,
+      address,
+      id,
+      path,
+      crypto: {
+        cipher: {
+          type: "aes-256-gcm",
```

Comment:
is there a technical reason we switched to aes-256-gcm? I would rather prefer XChaChaPoly as the default symmetric cipher mostly due to a rare but existing attack on aes-256-gcm: https://www.usenix.org/system/files/sec22summer_albertini.pdf

Reply:
Just for historical reference, `namada-interface` is actually using this package for encrypting stored keys: https://www.npmjs.com/package/redux-persist-transform-encrypt, and using `orion::aead` (which uses `XchaChaPoly1305`) only for the seed phrase (which is stored in `window.localStorage`). I've opened https://github.com/anoma/namada-interface/issues/124 to move our encrypted storage to use `XChaChaPoly`. In the extension, the stored mnemonic and private keys share the same encryption cipher (unlike how it is in `namada-interface` currently), and once the interface is loading accounts from the extension, it will no longer need to store any data itself. All accounts in the extension are stored in IndexedDB.

Reply:
can we keep somewhere an up to date document with the workflow of that protocol?Is that related with that? https://github.com/anoma/namada-interface/commit/85c80072468411950268b637a91cd6843d394162 and line 23? It is not very clear what is meant "encrypting stored keys".  Can we short describe it somewhere? as in the link above?